tests/test-frameworks/mixed/BUILD

```
@@ -3,15 +3,17 @@ load("@rules_scala_annex//rules:scala.bzl", "scala_test")
 scala_test(
     name = "mixed_2_13",
     srcs = glob(["*.scala"]),
+    deps_used_whitelist = [
```

Could you add a comment explaining this change? Might be worth a TODO to see if we can get the deps used to not mark these things required for test discovery as unused.

---

Are these not runtime deps? :thinking: If we don't depend on them but we need them it seems like they're runtime deps

---

They aren't, no. These tests aren't needed at runtime; they're needed for test discovery. The reason they have to be in `deps` now is because test discovery now happens at compile-time, not at runtime. I haven't updated the deps checker to mark these as used because in most cases, the target containing the implementation of the SBT test interface is the same one that the target would've depended on anyway. That's why none of the other external dependencies are in this list.

I can fix this, though. One important thing to note is that this will break a property of our dependency checker that's been thus-far guaranteed: that a dependency checker only allows a dependency to be in `deps` or not in `deps`. When we do test discovery, we go down a big list of test "frameworks". If we can find the test framework in the classpath, we look for tests that use that framework; otherwise, we don't. This means that, for example, if specs2 isn't on the classpath, but the target contains specs2 tests, the tests will silently fail to execute.

I think this is fine and unavoidable, but you disagree, I think we should leave this as-is.